SIP-X: Accepting ownership of Sovryn contracts

john-light opened the pull request:
Moved here: https://github.com/DistributedCollective/SIPS/pull/46

john-light committed:
SIP-X: Accepting ownership of Sovryn contracts

john-light committed:
Update SIP-X: Add contracts to be owned/administered

john-light committed:
Complete Motivation and Proposed Change sections

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Other             |                             |       |                                                              |
+|                   |	Liquidity Mining            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	VestingLogic                | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
same with the VestingLogic

john-light replied:
By "same" do you mean, as you said in your other comment, "These are irrelevant. The owner of the modules does not matter."?

john-light replied:
- Owner of the logic contract doesn't matter, only the owner of the proxy contract matters

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   |	Liquidity Mining            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	VestingLogic                | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Modules                     | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
what modules?

john-light replied:
This one: 

![Screenshot 2022-04-15 200616](https://user-images.githubusercontent.com/9424721/163653969-2571a5d5-a59b-4ada-a98d-3f23e88a7c43.png)

The admin role in line 38 is specifically referring to the `Protocol Settings` shown in the screenshot; upgrading the modules was assigned to `GovernorOwner` in lines 25-33.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   |	VestingLogic                | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Modules                     | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Connectors        |                             |       |                                                              |
+|                   |	LoanToken                   | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicBeacon        | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
For those we need to set the owner as well - to GovernorOwner

john-light replied:
So there is both an Admin and an Owner on these contracts? These weren't on the list of upgradeable smart contracts:
https://docs.google.com/document/d/1gGY4Rua_FVBZCJCftzf14cD4c6kqg6VTr9g-9-uDCA0/edit

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   |	LoanTokenSettingsLowerAdmin | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicStandard      | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Core              |	                            |     	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Protocol                    | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
we need to set the owner of the protocol as well (the protocol serves as proxy for the protocol modules)

john-light replied:
Done

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   |	Price Feeds Gateway         | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	PriceFeedV1PoolOracle       | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Governance        |	                            |      	|                                                              |
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
SOV is currently owned by GovernorOwner (or to be more precise: the timelock contract of GovernorOwner, which in turn is owned by GovernorOwner)
There is no admin role on the SOV contract

john-light replied:
Doesn't "owner" imply it can be upgraded? And I thought SOV was not upgradeable?

john-light replied:
`Owner` does not mean a contract is upgradeable, and SOV is not upgradeable.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | FeeSharingProxy             | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | StakingRewards              | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
staking rewards has no admin role, only owner

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Governance        |	                            |      	|                                                              |
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | FeeSharingProxy             | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
FeeSharingProxy has no admin role, only owner

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   | FeeSharingProxy             | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | StakingRewards              | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Origin Investor Claim       | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
This contract is no longer relevant. It was used only for the origins sale is of no use any longer.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | StakingRewards              | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Origin Investor Claim       | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Token Sender                | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
The Token Sender is a contract for improving internal processes.  When tokens need to be distributed to several different parties, exchequer can deposit the tokens to this contract and authorized addresses may then execute the transfers. This needs no governance involvement.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   | StakingRewards              | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Origin Investor Claim       | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Token Sender                | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Vesting Registry            | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
we can add the GovernorAdmin as an admin, but i'd like to point out that we need to keep the exchequer as an admin as well. Also, team members are temporarily admins of this contract in order to be able to add more vesting contracts to the registry. I would advise against requiring governance approval for each vesting contract.

john-light replied:
Makes sense. Who has the ability to add and remove admins?

john-light replied:
Admin can add and remove admins.

john-light replied:
Add Owner for Vesting Registry. Owner can:
- Intitialize contract
- Can set the Vesting Factory address
- Withdraw SOV from the Vesting Registry to any other address

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+|                   |	PriceFeedV1PoolOracle       | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Governance        |	                            |      	|                                                              |
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
that's fine, but this contract can have multiple admins. do we suggest here to merley add the GovernorAdmin as an admin or replace exchequer?
It would be fine to replace it in my opinion. The admin can change the cliff and duration used by the LockedSOV contract, add and remove more admins and migrate tokens to a new contract if required (the last 3 should have rather been placed under owner imo, but it is what it is)

john-light replied:
> this contract can have multiple admins.

do you mean multiple admins who all have the same capabilities? or multiple admins who each have different capabilities?

> do we suggest here to merley add the GovernorAdmin as an admin or replace exchequer?

Replace Exchequer, if it makes sense to you given the use/power of this contract and what we are trying to accomplish with this transfer of governance from Exchequer to Bitocracy. Btw - what exactly does the Locked SOV contract do? :)

> (the last 3 should have rather been placed under owner imo, but it is what it is)

Would it require a new contract to do this? Would you prefer that we change the `Admin` role to be held by GovernorOwner instead of GovernorAdmin in this case?

john-light replied:
Answered all of the above.

- All admins have the same capabilities
- Locked SOV is an "intermediate" contract between rewards contracts and the staking/vesting contract, after rewards are earned but before they are claimed/withdrawn by the user.
- Admin role will be assigned to GovernorAdmin for now.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Core              |	                            |     	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Protocol                    | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Oracles           |	                            |     	|                                                              |
+|                   |	BPro Price Feed             | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	RSK Price Feed              | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	MoC Price Feed              | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Price Feeds Gateway         | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	PriceFeedV1PoolOracle       | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
note: on the contracts there is only the role `owner`, but the owner has only administrative power, so it should be fine to place it under GovernorAdmin.

john-light replied:
Well, that is proper confusing xD thanks for letting me know

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+
+|	Category        	| Contract name	              | Role  | New governor                                                 |
+| ----------------- |:---------------------------:|:-----:|:------------------------------------------------------------:|
+| LoanToken Modules |	                            | 	    |                                                              |
+|                   |	LoanTokenLogicLM            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenLogicWRBTC         | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenSettingsLowerAdmin | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+| Protocol          |	                            |       |                                                              |
+|                   |	Affiliates                  | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsLiquidation     | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsRollover        | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsWith            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanMaintenance             | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanOpenings                | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanSettings                | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	ProtocolSettings            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	SwapsExternal               | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```
These are irrelevant. The owner of the modules does not matter. These logic contracts are just providing logic. Whenever the protocol or a loan token contract is executed, the storage of the proxy contract is used and therefore also the owner of the proxy contract. The owner of the modules has no power at all.

john-light replied:
FYI - You can comment on multiple lines by clicking on the number of the first line you want to comment on and dragging it down to the last line you want to comment on :) With that having been said, which lines are you referring to here?

> Whenever the protocol or a loan token contract is executed, the storage of the proxy contract is used and therefore also the owner of the proxy contract. 

What does it mean for "the protocol or a loan token contract.." to be "executed"? What does it mean for "the storage of the proxy contract" to be "used and therefore also the owner of the proxy contract"? I'm having a hard time understanding this given my (lacking of?) knowledge of upgradeable contracts.

> The owner of the modules has no power at all.

Is there even an owner for these modules defined? If so, why, if they have no power? Would also repeat my [other question](https://github.com/john-light/sovryn/pull/7/files#r851281231) about the owner role here.

john-light replied:
- Storage on the logic contracts isn't used. Only the storage on the proxy is used.

ororopickpocket commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Connectors        |                             |       |                                                              |
+|                   |	LoanToken                   | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicBeacon        | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenSettingsLowerAdmin | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicStandard      | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
again irrelevenat, because always called through the LoanToken contract which serves as proxy

john-light replied:
So the admin of LoanToken can call `Set Liquidity Mining address of the LoanToken` on the LoanTokenLogicStandard connector contract? 

Trying to parse how these connect together, looking at https://docs.google.com/document/d/1DoTxAlOFE27HZ7BPcMv_wHX8scqmVwjpKTxlcdeuvHI/edit

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Core              |	                            |     	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Protocol                    | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Oracles           |	                            |     	|                                                              |
+|                   |	BPro Price Feed             | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	RSK Price Feed              | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	MoC Price Feed              | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Price Feeds Gateway         | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	PriceFeedV1PoolOracle       | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
|                   |	BPro Price Feed             | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
|                   |	RSK Price Feed              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
|                   |	MoC Price Feed              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
|                   |	Price Feeds Gateway         | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
|                   |	PriceFeedV1PoolOracle       | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```

john-light committed:
Admin -> Owner

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,75 @@
+| Governance        |	                            |      	|                                                              |
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | FeeSharingProxy             | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove FeeSharingProxy change

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,74 @@
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | StakingRewards              | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove StakingRewards change

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,73 @@
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Origin Investor Claim       | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove Origin Investor Claim change

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,72 @@
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Token Sender                | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove Token Sender change

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,71 @@
+Furthermore, this proposal will result in a signal from SOV stakers that they are willing and ready to accept the owner and/or adminstrator role in the following Sovryn smart contracts, to be transferred by the Exchequer Multisig no later than `2022-XX-XX 23:59:59 UTC`:
+
+|	Category        	| Contract name	              | Role  | New governor                                                 |
+| ----------------- |:---------------------------:|:-----:|:------------------------------------------------------------:|
+| LoanToken Modules |	                            | 	    |                                                              |
+|                   |	LoanTokenLogicLM            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenLogicWRBTC         | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenSettingsLowerAdmin | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+| Protocol          |	                            |       |                                                              |
```
test

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,71 @@
+
+|	Category        	| Contract name	              | Role  | New governor                                                 |
+| ----------------- |:---------------------------:|:-----:|:------------------------------------------------------------:|
+| LoanToken Modules |	                            | 	    |                                                              |
+|                   |	LoanTokenLogicLM            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenLogicWRBTC         | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanTokenSettingsLowerAdmin | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+| Protocol          |	                            |       |                                                              |
+|                   |	Affiliates                  | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsLiquidation     | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsRollover        | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanClosingsWith            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanMaintenance             | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanOpenings                | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanSettings                | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	ProtocolSettings            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	SwapsExternal               | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```
```suggestion
```

john-light committed:
Remove logic contracts

- Logic storage is irrelevant

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,57 @@
+| Other             |                             |       |                                                              |
+|                   |	Liquidity Mining            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	VestingLogic                | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove vesting logic

- Owner of logic contract is irrelevant

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,56 @@
+Furthermore, this proposal will result in a signal from SOV stakers that they are willing and ready to accept the owner and/or adminstrator role in the following Sovryn smart contracts, to be transferred by the Exchequer Multisig no later than `2022-XX-XX 23:59:59 UTC`:
+
+|	Category        	| Contract name	              | Role  | New governor                                                 |
+| ----------------- |:---------------------------:|:-----:|:------------------------------------------------------------:|
```
```suggestion
| ------------------ |:---------------------------:|:-----:|:------------------------------------------------------------:|
| Core |:---------------------------:|:-----:|:------------------------------------------------------------:|
| -------- | Protocol | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,56 @@
+| Other             |                             |       |                                                              |
+|                   |	Liquidity Mining            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	Modules                     | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light replied:
"Modules" is a folder, not a contract

john-light committed:
Remove "Modules"

"Modules" is a folder, not a contract

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,55 @@
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Locked SOV                  | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Staking                     | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   | Vesting Registry            | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
|                   | Vesting Registry             | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
|                   | Vesting Registry            | Admin | Exchequer Multisig (`0x924f5ad34698Fd20c90Fe5D5A8A0abd3b42dc711`) |
```

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,55 @@
+|	Category        	| Contract name	              | Role  | New governor                                                 |
+| ----------------- |:---------------------------:|:-----:|:------------------------------------------------------------:|
+| Other             |                             |       |                                                              |
+|                   |	Liquidity Mining            | Owner	| GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```
```suggestion
| | Liquidity Mining | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
| | Vesting Registry | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```

john-light committed:
Add vesting registry owner

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,56 @@
+| | Liquidity Mining | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+| | Vesting Registry | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Connectors        |                             |       |                                                              |
```
```suggestion
| Connectors         |                             |       |                                                              |
| | LoanTokenLogicBeacon | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
```

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,56 @@
+| | Vesting Registry | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	StakingRewards              | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Connectors        |                             |       |                                                              |
+|                   |	LoanToken                   | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicBeacon        | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
|                   |	LoanToken                   | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
|                   |	LoanToken                   | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```

john-light committed:
Add LoanToken owner

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,56 @@
+| Connectors        |                             |       |                                                              |
+|                   |	LoanToken                   | Owner | GovernorOwner (`0x6496df39d000478a7a7352c01e0e713835051ccd`) |
+|                   |	LoanToken                   | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenSettingsLowerAdmin | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	LoanTokenLogicStandard      | Admin	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove irrelevant contracts

john-light commented on SIP-X.md:
```diff
@@ -0,0 +1,54 @@
+|                   |	Price Feeds Gateway         | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+|                   |	PriceFeedV1PoolOracle       | Owner	| GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
+| Governance        |	                            |      	|                                                              |
+|                   | SOV                         | Admin | GovernorAdmin (`0xff25f66b7d7f385503d70574ae0170b6b1622dad`) |
```
```suggestion
```

john-light committed:
Remove SOV admin

john-light committed:
Add Protocol owner

john-light committed:
Update SIP-X.md

Add LoanTokenLogicBeacon Owner

john-light committed:
Add Vesting Registry admin

- Also specify that we are keeping Exchequer Multisig as an admin on Vesting Registry contract

john-light committed:
Update table formatting
